pkg/runtime/interfaces.go

```diff
@@ -20,29 +20,23 @@ import (
 	"io"
 )
 
-// ObjectScheme represents common conversions between formal external API versions
-// and the internal Go structs. ObjectScheme is typically used with ObjectCodec to
-// transform internal Go structs into serialized versions. There may be many valid
-// ObjectCodecs for each ObjectScheme.
-type ObjectScheme interface {
-	ObjectConvertor
-	ObjectTyper
-	ObjectCreater
-	ObjectCopier
-}
-
-// ObjectCodec represents the common mechanisms for converting to and from a particular
-// binary representation of an object.
-type ObjectCodec interface {
-	ObjectEncoder
+// Codec defines methods for serializing and deserializing API objects.
+type Codec interface {
 	Decoder
+	Encoder
+
+	// Methods for processing url parameters.
+	// EncodeParameters(obj Object) (url.Values, obj Object) error
+	// DecodeParametersInto(parameters url.Values, obj Object) error
```

Comment:
Hm, now that I think about it some more, maybe it doesn't matter.

But if you don't pass an object in, then there has to be some generic way for the codec to know what the context is. E.g., "?resourceVersion=1234" could be a valid ListOptions or a valid DeleteOptions (maybe). But the strategy can solve that by making an appropriate options object. E.g.,

```go
options := strategy.NewOptions()
err := codec.DecodeParametersInto(myurl.Query, options)
strategy.Method(options, body)
```

If we don't do this, we'd probably have to pass in the verb, which is not terrible. But this way we keep the decision about the type of the options object near the consumption of the options object, which seems good.